micro-rdk/src/esp32/tls.rs

```diff
@@ -199,17 +207,16 @@ impl Esp32TLSStream {
                 "couldn't allocate tls context",
             ));
         }
-        let tls_context = ManuallyDrop::new(p);
+        let tls_context = TLSContext(ManuallyDrop::new(p));
```

Comment:
Could the `TLSContext` wrapper just take care of calling `esp_tls_init`? Then there would be no window where it could go wrong. As written, `p` comes into existence at line 203 but it isn't truly under proper management until after 210.

I think it would mostly amount to just adding a `TLSContext::new() -> Result<Self, SomeErrorHere>` and then calling it like `let tls_context = TLSContext::new()?` here.

Reply:
yeah that's fair, I'll add that